src/libsyntax/parse/lexer/mod.rs

```diff
@@ -1367,64 +1319,124 @@ impl<'a> StringReader<'a> {
         id
     }
 
-    fn scan_raw_byte_string(&mut self) -> (token::LitKind, Symbol) {
-        let start_bpos = self.pos;
+    fn scan_raw_string(
+        &mut self,
+        start_bpos: BytePos,
+        raw_type: RawStringType
+    ) -> (token::LitKind, Symbol) {
         self.bump();
-        let mut hash_count = 0;
+        let mut hash_count: u16 = 0;
         while self.ch_is('#') {
             if hash_count == 65535 {
                 let bpos = self.next_pos;
                 self.fatal_span_(start_bpos,
                                  bpos,
-                                 "too many `#` symbols: raw byte strings may be \
+                                 "too many `#` symbols: raw strings may be \
                                  delimited by up to 65535 `#` symbols").raise();
             }
             self.bump();
             hash_count += 1;
         }
-
-        if self.is_eof() {
-            self.fail_unterminated_raw_string(start_bpos, hash_count);
-        } else if !self.ch_is('"') {
-            let pos = self.pos;
-            let ch = self.ch.unwrap();
-            self.fatal_span_char(start_bpos,
-                                        pos,
-                                        "found invalid character; only `#` is allowed in raw \
-                                         string delimitation",
-                                        ch).raise();
+        let bpos_span = self.mk_sp(start_bpos, self.pos);
+
+        match self.ch {
+            None => self.fail_unterminated_raw_string(bpos_span,
+                                                      hash_count,
+                                                      vec![self.mk_sp(self.pos, self.pos)]),
+            Some('"') => {},
+            Some(c) => {
+                let last_bpos = self.pos;
+                self.fatal_span_char(start_bpos,
+                                     last_bpos,
+                                     "found invalid character; only `#` is allowed \
+                                      in raw string delimitation",
+                                     c).raise();
+            }
         }
+
         self.bump();
         let content_start_bpos = self.pos;
         let mut content_end_bpos;
+        let mut valid = true;
+        let mut spans = vec![];
+
         'outer: loop {
-            match self.ch {
-                None => {
-                    self.fail_unterminated_raw_string(start_bpos, hash_count);
-                }
-                Some('"') => {
+            match (self.ch, raw_type) {
+                (None, _) => {
+                    spans.push(self.mk_sp(self.pos, self.pos));
```

Comment:
I don't think the suggestion for this span will ever be valid.

Reply:
Maybe you're right. This was just an attempt to make a suggestion if the file ends and the raw string hasn't ended yet. I can remove that.

Reply:
Yeah, I see where you were getting at, but it is unlikely that the problem is ever, and if it is, the solution is obvious but we cannot use rustfix to fix it.
```
let x = r##"
```